pkg/trace/info/stats.go

```diff
@@ -288,6 +284,46 @@ func (s *SpansMalformed) String() string {
 	return mapToString(s.tagValues())
 }
 
+const (
+	maxAbsSamplingPriority = 10
+	numPriorityCounters    = maxAbsSamplingPriority*2 + 1
+)
+
+type SamplingPriorityStats struct {
+	// tracesPerSamplingPriority holds counters for each priority in position MaxAbsPriorityValue + priority.
+	tracesPerSamplingPriority [numPriorityCounters]int64
+}
+
+func (s *SamplingPriorityStats) CountSamplingPriority(priority sampler.SamplingPriority) {
+	if priority < maxAbsSamplingPriority && priority > (-1*maxAbsSamplingPriority) {
+		atomic.AddInt64(&s.tracesPerSamplingPriority[maxAbsSamplingPriority+priority], 1)
```

Comment:
What happens if we’re writing a map entry while another goroutine is reporting stats from it? Is that not a potential race condition?